packages/apollo-react-relay-duct-tape/src/storeObservation/compiledHooks/useCompiledFragment.ts

```diff
@@ -28,73 +29,102 @@ export function useCompiledFragment(
     "useFragment(): Expected a `watchQueryDocument` to have been " +
       "extracted. Did you forget to invoke the compiler?",
   );
+  const from = fragmentReference.id;
+  invariant(
+    typeof from === "string",
+    "useFragment(): Expected the fragment reference to have a string id. " +
+      "Did you forget to invoke the compiler?",
+  );
```

Comment:
I don't think we can do it here, because `fragmentReference.id` can be `undefined`, which is expected according to comments here:
 
![image](https://github.com/user-attachments/assets/518c1930-22a2-4fa5-bd98-c2854743c9b4)

Reply:
`fragmentReference.id` is required for useFragment args to connect fragment with a query. @alloy do you think we can accept `fragmentReference.id` is nillable? Previous implementation did not have such invariant, but cache.watch did not require it either